widget/textgrid.go

```diff
@@ -47,9 +47,15 @@ type TextGridStyle interface {
 	BackgroundColor() color.Color
 }
 
+// TextGridTextStyle defines the text style that can be applied to a TextGrid cell.
+type TextGridTextStyle interface {
+	TextStyle() *fyne.TextStyle
+}
+
 // CustomTextGridStyle is a utility type for those not wanting to define their own style types.
 type CustomTextGridStyle struct {
 	FGColor, BGColor color.Color
+	Text             *fyne.TextStyle
```

Comment:
Wouldn't `Style` or `TextStyle` be more appropriate than `Text` for a style field?
Perhaps it should match naming elsewhere like on the `canvas.Text` or `widget.Label`?

p.s. It's missing the `// Since: 2.5` comment

Reply:
I've replaced this with two bool values

Reply:
Taking this path means that adding another value later will need another interface... Using TextStyle (which is a concrete struct) would make it possible to add more I think

Reply:
Text Style would need the underline property adding.

Reply:
Indeed. But do you think underline is only useful for TextGrid? Or would we perhaps consider having it broadly available under a simple style value?

Reply:
No, it could/would make sense. Just making you aware of the implications. I'm finding it very dificult to determine the scope of changes to fyne and associated projects as in some cases you suggest less invasive and some other cases more invasive. The implications of adding this to TextStyle would mean it would remain unimplemented by anything that uses it and other options like italics which will not be implemented within the TextGrid/Renderer.